0.4.0

anovis opened the pull request:
### Version 0.4.0

### **New Features**

- CLI

### **New Particles**

**Azure**
- azure resource
- blob storage

### **New Quasiparticles**

- Cross Cloud Storage

### **Improvements**
- logging in quasiparticle
- just-in-time-infrastructure example
- renamed few particles to better reflect naming conventions 
- set desired_state with string instead of State
- concept section in documentation
- added style guide

anovis committed:
Merge pull request #1 from capitalone/master

merge master

anovis committed:
auto generate zip for python lambda

anovis committed:
fix lambda flavor

anovis committed:
skip packages that are not found

anovis committed:
fixed nested directory bug in zip file

anovis committed:
ability to set a different requirements.txt name to list python packages. default is still requirement.txt

anovis committed:
Update requirements.txt

awscli is required  by moto if credential files is found.

anovis committed:
Merge pull request #26 from anovis/bug-moto-test

Bug moto test

emr0327 committed:
Pcf max timeout in apply() (#8)

* configured max_timeout feature in pcf class and quasiparticle

* added max_timeout feature unit test for quasiparticle

* Applied the assertion check for test_passing_vars() test case for the InvalidValueReplaceException thrown in quasiparticle.apply()

* Quasiparticle raises particle errors

anovis committed:
add iam to particle list in docs (#28)

davidyum committed:
Iam role (#32)

* update iam role to take in instanceprofile flag

* examples to include IsInstanceProfile flag

* test to check instance profile was created

eloucks committed:
Adding KMS key particle + tests (#37)

* KMS key particle + tests

* Added example

emr0327 committed:
Merge pull request #14 from anovis/lambda_extension

Lambda extension

brianhlee committed:
Cloudfront particle (#27)

* cloudfront particle

particle class, test, example, and docs

anovis committed:
removed botocore dependency. 

removed botocore dependency.

anovis committed:
Just in Time Infrastructure example (#34)

* JIT QP

* fixed bug for str in list for finding and replacing

davidyum committed:
Flask app (#35)

* add readme and sample json

* dockerize flask app

brianhlee committed:
Auto generate test (#36)

* auto generate test for particle classes (3 examples in this pr)

* add placebo and moto mocking into the test generator

* fixed update_dict

brianhlee committed:
renamed aws resouces to follow naming conventions (#49)

* renamed aws resouces to follow naming conventions

hosted zone was fine as is
vpc -> vpc instance
compute -> compute engine
storage -> bucket

davidyum committed:
Pass list var (#51)

* added test for passing list of variables from parent to child using inherit

jacoblearned committed:
Add CLI (#38)

* gitignore venv directories from virtualenv

* Add color and failure utils for CLI

* Update to boto3 v1.9.76

This prevents a version conflict with the version of botocore we were using

* Move pcf version to importable variable

* Add top-level pcf command

* Add suggestion for command typos

Uses Levenshtein distance against list of supported commands specified as PCFCLI class variable

* Add list_commands method to PCFCLI

This is used by Click to list the commands in help prompts

* Add apply command placeholder

* Add help when just 'pcf' is the command

* Add tests for top-level pcf command

* Update awscli to 1.16.84

* Remove unused module in CLI

* Add pyyaml to required packages

* Add did_you_mean utility function

Helps with printing potentially intended string values

* Add class-finding utilty from a particle flavor

* Add utility functions for reading pcf config files

* Moves compute client to constructor of ComputeEngine

The previous position will execute the Google Authentication Code, which may cause unexpected errors when performing metaprogramming for tasks such as loading the class from the CLI

* Fix particle class search and clean error msgs

* Normalize pcf config to list when reading from file

* Add Particle class instantiation to CLI

Takes a target pcf_name value and searches a config file for the specified particle config, suggesting potentially close typos if the particle class or pcf_name isn't found

* Add basic apply command execution

Attempts to load the particle class, set the desired state based on the value of the state flag (defaulting to State.running) and then runs apply on the particle

* Add pcf run command

Convenience comamnd equivalent to
`pcf apply my_ec2 --state running`

* Add pcf stop command

* Add pcf terminate command

* Add utility function for executing an applying command

* Add pytest fixture for copying test PCF config files

* Add test pcf config files

* Add tests for apply command

* Add tests for run command

* Add tests for stop command

* Add tests for terminate command

* Format command test code

* Format run, stop, and terminate commands

* Improve help messages for commands

* Format stop and terminate commands

* Combine no_color and color utility functions

* Compare levenshtein distance with lowercase strings

* Add tests for color, fail, and similar_strings util functions

* Use basename for output in when loading pcf config

* Simplify pcf config file loading interface

* Add ill-formed pcf config file fixtures

* Improve error msgs when loading particles from file

* Add useful test deployments to pcf.yml

* Change silent option to quiet

* Add -q or --quiet flag to apply, run, stop, and terminate

* Add tests for utility functions

* Add -c or --cascade option for applying with cascade=True

* Scope pytest testpath to pcf/test directory

* Handle import errors when searching for particle class

* Add test for import errors in submodule lookup

* Add make test task w/ code coverage

Adds a .coveragerc to ignore the tests/directory when generating coverage reports

* Add tests for missing LoC

Tests bare `pcf` command and non-dict entries in a pcf config file list

* Add click_options util to extract common CLI options

* Add max timeout common CLI option

* Move particle_class_from_flavor to pcf core utils

* Include flavor name in loading message

* Make pcf_name an optional argument

If the pcf_name argument isn't given to an applying command, all apply() will be called on all particles

* Clean applying cmd options and help messages

* Add progress bar for applies to >1 particle

* Remove "changes" from command help strs

* Update default version to  0.3.2

* Add doc for CLI usage

anovis committed:
pcf core improvements (#55)

improvement for the STATE_STRING_TO_ENUM

emr0327 committed:
PCF Style Guide (#6)

* added style guide for PCF

anovis committed:
Jitidocs (#50)

* added section core concepts with entry on JiTI
* added note on python version requirement.

anovis committed:
change logging type and message in quasiparticle apply (#59)

brianhlee committed:
Azure storage (#45)

* azure resource and blob storage particle

anovis committed:
Merge branch 'master' into develop

brianhlee commented:
can we merge the name refactoring branch first? there are a few errors in the docs

brianhlee committed:
Name refactoring (#53)

* renamed aws resouces to follow naming conventions

didn't know what to name the gcp resource, so left them as is for now

* edit

hosted zone was fine as is
vpc -> vpc instance
compute -> compute engine
storage -> bucket

* change vpc class name

* fixed changes that I missed in VPCInstance

flavor name
and one place i missed for changing the class name

* docs fixes

applied the name-refactoring to the docs

changed the subsubsections (particles) in the particle list to be different from the sub sections (aws, az, gcp)

davidyum committed:
Cross cloud storage (#44)

* create cross cloud storage quasiparticle
* add docs for cross cloud storage

anovis committed:
Set desired state with string (#62)

* can set state with string

eloucks commented:
Should we merge in the CI before the release?

ethanwlo reviewed (approved):
lgtm
